bibliography.bib

```diff
@@ -7,6 +7,68 @@ @inproceedings{gullasch2011cache
+    Once co-location is achieved, an attacker can use side-channels (namely caches) to get information about the other instances running on 
+    the same machine as the malicious VM.  
+    \par
+    A 'map' of the EC2 service was constructed in order improve the likelihood of placing a VM on the same host as the target. 
+    By creating multiple instances for each permutation of the creation parameters, the authors observed a trend between internal IP addresses
+    and creation parameters. 
+    These internal IP addresses were concluded to be statically associated with physical machines. 
+    Use of the created map requires translating the external IP of the target to an internal IP, which is made possible by the DNS service
+    provided by EC2.
+    \par
+    There are multiple ways to check if two instances are co-resident. 
+    Each physical machine has a privileged VM associated with it called Domain0, or Dom0.
+    Two VMs which are co-located will have identical Dom0 IP addresses and internal IP addresses which are numerically close.
+    Measuring the round-trip time for a packet sent between the two VMs can also be used to tell if they are co-resident, as it is 
+    significantly lower in co-resident instances.
+    The Dom0 IP check was used for all subsequent co-location testing as experiments showed that it produced no false positives.
+    \par
+    Two strategies were used to prove that achieving co-residence was possible. 
+    The first used a brute-force technique by creating a large number of probe instances and targeting a large number of potential victims.
+    The probes checked if they were co-resident with any of the target VMs, then were immediately deactivated if they were not.
+    This method achieved a reasonable success rate, covering 8.4\% of the targets. 
+    A second method monitored a victim server, waiting for it to disappear when not needed, then reappear as a new instance. 
+    The attacker then ran many instances using the same creation parameters as the target. 
+    Experiments with the placement of new instances found parallel placement locality (instances launched at around the same time tend to be 
+    placed on the same hardware). 
+    This, along with the 'map' created earlier, led to a much more successful approach, achieving co-residence 40\% of the time.
+    \par
+    Finally, the authors describe attacks which can be carried out on a co-resident VM. 
+    Most of these attacks take advantage of the \name{Prime+Probe} technique \cite{osvik2006cache, tromer2009cache} 
+    to measure the load on a given machine.
+    By measuring cache usage, an attacker can determine co-location without using the techniques described earlier. 
+    Cache usage can also be used to estimate traffic to a web server or measure the length of time between keystrokes for the purpose of 
+    recovering a password.
+  }
+  
```

Comment:
I just noticed that this entry is never closed. BibTeX generates an error, but the system still manages to make the document.